File: rabbitmq/publisher.py
#!/usr/bin/env python3
import pika


def publisher(channel, i):
    channel.basic_publish(body='This is message # {}'.format(i), routing_key='foo', exchange='')


parameters = pika.URLParameters('amqp://guest:guest@localhost:5672/%2F')
connection = pika.BlockingConnection(parameters)
chan = connection.channel()
chan.queue_declare(queue='foo')

for i in range(10000000):
    publisher(chan, i)

File: rabbitmq/publisher.pl
#!/usr/bin/perl
#
use Net::AMQP::RabbitMQ;
my $mq = Net::AMQP::RabbitMQ->new();
$mq->connect("localhost", { user => "guest", password => "guest" });
$mq->channel_open(1);
$mq->queue_declare(1, "foo");

my $count = 0;
while(1) {
    $mq->publish(1, "foo", sprintf("Test message # %d", $count));
    $count++;
}
$mq->disconnect();
